src/SFML/Audio/SoundRecorder.cpp

```diff
@@ -59,7 +60,12 @@ m_deviceName        (getDefaultDevice())
 ////////////////////////////////////////////////////////////
 SoundRecorder::~SoundRecorder()
 {
-    // Nothing to do
+    // This assertion is triggered if the recording is still running while
+    // the object is destroyed. It ensures that stop() is called in the
+    // destructor of the derived class, which makes sure that the recording
+    // thread finishes before the derived object is destroyed. Otherwise a
+    // "pure virtual method called" exception is triggered.
```

Comment:
Its true that there is some duplication, but I think in this case it should stay in. This is error is definitly not trivial. I think the comment explains a little more detailed whats happening here any also why. It makes it easier to understand for future readers what kind of error is resolved here. If you read the discussion above this comment was specifically asked for.